lib/ProductOpener/Text.pm

```diff
@@ -47,6 +47,18 @@ sub normalize_percentages($$) {
 
 	my ($text, $locale) = @_;
 
+	# Bail out of this function if no known percent sign is found.
+	# This is purely for performance reasons: CLDR functions are
+	# comparatively expensive to run.
+	if ((not (defined $text))
+		or (not ((index($text, "\N{U+0025}") > -1)
+		or (index($text, "\N{U+066A}") > -1)
+		or (index($text, "\N{U+FE6A}") > -1)
+		or (index($text, "\N{U+FF05}") > -1)
+		or (index($text, "\N{U+E0025}") > -1)))) {
```

Comment:
This is a bit more difficult to read than
```
		or ((index($text, "\N{U+0025}") == -1)
		and (index($text, "\N{U+066A}") == -1)
		and (index($text, "\N{U+FE6A}") == -1)
		and (index($text, "\N{U+FF05}") == -1)
		and (index($text, "\N{U+E0025}") == -1))) {
```
but as it performs a bit fewer `index` calls, the avg in my test runs was 4µs vs. 3µs.